### PR TITLE
{Core} Copy login credential when run test with `--live` and `random_config_dir` is enabled 

### PR DESCRIPTION
**Pull request by bebound**
**Description**

As mentioned in https://github.com/Azure/azure-cli/pull/26273#issuecomment-1551174298, the login info can't be retrieved in recording mode when enable `random_config_dir`,
This PR copies the login credential from `~/.azure` to random config dir.

Related PR: https://github.com/Azure/azure-cli/pull/25689

**Testing Guide**

`azdev test test_tag_update_by_patch --live`
Before:
`FAILED src\azure-cli\azure\cli\command_modules\resource\tests\latest\test_resource.py::TagScenarioTest::test_tag_update_by_patch - knack.util.CLIError: Please run 'az login' to setup account.`
After:
`Message: The client 'xxx' with object id 'xxx' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/write' over scope '/subscriptions/axx' or the scope is invalid. If access was recently granted, please refresh your credentials.`
So it can read the token.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/mock.py [hunk 1]
```diff
@@ -17,16 +17,23 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
...
 
         from knack.completion import ARGCOMPLETE_ENV_NAME
 
+        # random_config_dir only takes effects in playback mode
+        if random_config_dir and os.getenv(ENV_VAR_TEST_LIVE, '').lower() != 'true':
+            config_dir = os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir', random_string())
+            self.use_random_config_dir = True
+        else:
+            config_dir = GLOBAL_CONFIG_DIR
+            self.use_random_config_dir = False
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/mock.py:32]: Won't this trigger failure when executed concurrently in live mode?

Comment (bebound) [reply src/azure-cli-core/azure/cli/core/mock.py:32]: That's a valid point. I need to copy login credentials then.

File: src/azure-cli-core/azure/cli/core/mock.py [hunk 2]
```diff
@@ -17,16 +17,33 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
...
+        # random_config_dir only takes effects in playback mode
+        if random_config_dir:
+            config_dir = os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir', random_string())
+
+            # In recording mode, copy login credentials from global config dir to the dummy config dir
+            if os.getenv(ENV_VAR_TEST_LIVE, '').lower() == 'true':
+                if os.path.exists(GLOBAL_CONFIG_DIR):
+                    ensure_dir(config_dir)
+                    import shutil
+                    for file in ['azureProfile.json', 'msal_token_cache.bin', 'clouds.config']:
```
Comment (bebound) [inline src/azure-cli-core/azure/cli/core/mock.py:36]: @jiasli Are these files requires as login credential?

File: src/azure-cli-core/azure/cli/core/mock.py [hunk 3]
```diff
@@ -17,16 +17,33 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
...
+                        try:
+                            shutil.copy(os.path.join(GLOBAL_CONFIG_DIR, file), config_dir)
+                        except FileNotFoundError:
+                            pass
```
Comment (zhoxing-ms) [inline src/azure-cli-core/azure/cli/core/mock.py:40]: I am not very familiar with the knowledge in the field of auth, may I ask why do we pass for the `FileNotFoundError`?

Comment (bebound) [reply src/azure-cli-core/azure/cli/core/mock.py:40]: If the user is not logged in yet, these files do not exist in `~/.azure`. Skip copying them.

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**